web/api/v1/api.go

```diff
@@ -570,14 +570,12 @@ func (api *API) labelValues(r *http.Request) (result apiFuncResult) {
 // LabelValuesByMatchers uses matchers to filter out matching series, then label values are extracted.
 func labelValuesByMatchers(sets []storage.SeriesSet, name string) ([]string, storage.Warnings, error) {
 	set := storage.NewMergeSeriesSet(sets, storage.ChainedSeriesMerge)
-	labelValuesMap := make(map[string]string)
+	labelValuesSet := make(map[string]struct{})
 	for set.Next() {
 		series := set.At()
 		labelValue := series.Labels().Get(name)
-		// This is to avoid duplicates.
-		_, ok := labelValuesMap[labelValue]
-		if !ok && labelValue != ""  {
-			labelValuesMap[labelValue] = labelValue
+		if labelValue != ""  {
```

Comment:
The label value can't be empty, if it is that's a bug in the tsdb.

Reply:
fixed